osquery/tables/system/windows/processes.cpp

```diff
@@ -41,107 +42,111 @@ std::set<long> getProcList(const QueryContext& context) {
-      r["user_time"] = BIGINT(llHolder / 10000000);
-      results[0].GetString("KernelModeTime", sPlaceHolder);
-      osquery::safeStrtoll(sPlaceHolder, 10, llHolder);
-      r["system_time"] = BIGINT(llHolder / 10000000);
-
-      results[0].GetString("PrivatePageCount", sPlaceHolder);
-      r["wired_size"] = BIGINT(sPlaceHolder);
-      results[0].GetString("WorkingSetSize", sPlaceHolder);
-      r["resident_size"] = sPlaceHolder;
-      results[0].GetString("VirtualSize", sPlaceHolder);
-      r["total_size"] = BIGINT(sPlaceHolder);
-      results_data.push_back(r);
-    }
+  result.GetString("Name", r["name"]);
+  result.GetString("ExecutablePath", r["path"]);
+  result.GetString("CommandLine", r["cmdline"]);
+  result.GetString("ExecutionState", r["state"]);
+  result.GetLong("ParentProcessId", lPlaceHolder);
+  r["parent"] = BIGINT(lPlaceHolder);
+  result.GetLong("Priority", lPlaceHolder);
+  r["nice"] = INTEGER(lPlaceHolder);
+  r["on_disk"] = osquery::pathExists(r["path"]).toString();
+
+  std::vector<char> fileName(MAX_PATH);
+  fileName.assign(MAX_PATH + 1, '\0');
+  if (pid == currentPid) {
+    GetModuleFileName(nullptr, fileName.data(), MAX_PATH);
+  } else {
+    GetModuleFileNameEx(hProcess, nullptr, fileName.data(), MAX_PATH);
+  }
+  r["cwd"] = SQL_TEXT(fileName.data());
+  r["root"] = r["cwd"];
+
+  r["pgroup"] = "-1";
+  r["uid"] = "-1";
+  r["euid"] = "-1";
+  r["suid"] = "-1";
+  r["gid"] = "-1";
+  r["egid"] = "-1";
+  r["sgid"] = "-1";
+  r["start_time"] = "0";
+
+  // We pre-populate the info, in the event the API calls fail.
+  result.GetString("UserModeTime", sPlaceHolder);
+  long long llHolder;
+  osquery::safeStrtoll(sPlaceHolder, 10, llHolder);
+  r["user_time"] = BIGINT(llHolder / 10000000);
+  result.GetString("KernelModeTime", sPlaceHolder);
+  osquery::safeStrtoll(sPlaceHolder, 10, llHolder);
+  r["system_time"] = BIGINT(llHolder / 10000000);
+
+  result.GetString("PrivatePageCount", sPlaceHolder);
+  r["wired_size"] = BIGINT(sPlaceHolder);
+  result.GetString("WorkingSetSize", sPlaceHolder);
+  r["resident_size"] = sPlaceHolder;
+  result.GetString("VirtualSize", sPlaceHolder);
+  r["total_size"] = BIGINT(sPlaceHolder);
+  results_data.push_back(r);
+
+  // OpenProcess returns nullptr on error, GetCurrentProcess returns a
+  // psuedo-handle of -1, which is the same as INVALID_HANDLE_VALUE
+  if (hProcess != nullptr && hProcess != INVALID_HANDLE_VALUE) {
+    CloseHandle(hProcess);
   }
 }
 
 QueryData genProcesses(QueryContext& context) {
   QueryData results;
 
-  auto pidlist = getProcList(context);
-  for (const auto& pid : pidlist) {
-    genProcess(pid, results);
+  std::string query = "SELECT * FROM Win32_Process";
+
+  auto pidlist = getSelectedPids(context);
+  if (pidlist.size() > 0) {
+    std::vector<std::string> constraints;
+    for (const auto& pid : pidlist) {
+      constraints.push_back("ProcessId=" +
+                            boost::lexical_cast<std::string>(pid));
```

Comment:
Sorry I didn't see this before, just a nit/note mostly, std::to_string is much faster than boost's lexical cast.